azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py

```diff
@@ -0,0 +1,202 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+# NOTE: The version uodate rules in this doc is complied with the doc below
+# https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md
+
+from packaging.version import parse
+from ._const import (PREVIEW_INIT_SUFFIX, VERSION_MAJOR_TAG, VERSION_MINOR_TAG,
+                     VERSION_PATCH_TAG, VERSION_STABLE_TAG, VERSION_PREVIEW_TAG, VERSION_PRE_TAG,
+                     CLI_EXTENSION_INDEX_URL)
+from .utils import DiffLevel
+
+
+class ModuleVersion:
+
+    def __init__(self, parse_version):
+        self.major = parse_version.major
+        self.minor = parse_version.minor
+        self.patch = parse_version.micro
+        self.pre = parse_version.pre
+        self.pre_num = parse_version.pre and parse_version.pre[1]
+
+    def init_stable_version(self):
+        self.major = 1
+        self.minor = 0
+        self.patch = 0
+        self.pre = None
+        self.pre_num = 0
+
+    def init_preview_version(self):
+        self.major = 1
+        self.minor = 0
+        self.patch = 0
+        self.pre = "b"
+        self.pre_num = 1
+
+    def __str__(self):
+        version_arr = [self.major, ".", self.minor, ".", self.patch]
+        if self.pre:
+            version_arr.append(self.pre)
+            version_arr.append(self.pre_num)
+        return "".join([str(item) for item in version_arr])
+
+
+class VersionUpgradeMod:
```

Comment:
I think version upgrade code should be in `azdev` instead of `azure-cli-diff-tool`.

Reply:
Version upgrade is working based on diff tool, and in extension ci, version calculation is planned to be used as func, so packaged with meta-diff together. azdev can expose related cmd later if needed.